### PR TITLE
Health scaners announces diseases on medical channel!

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
The health analyzer, adv health analyzer, and cyborg health analyzer now send out warnings on the Medical radio, as well as modifies medical records to display the viruses scanned.

It also has a cooldown now, so after scanning, it wont transmit a radio message for another 20 seconds

## Why It's Good For The Game
Gives medical more "official" warning if theres a disease, allows for anyone using a scanner to alert medical if a disease is found, gives an actual use for medrecords

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

medscan_cpu

medscan_scan

</details>

## Changelog
:cl:
add: Handheld medical analyzers now announce diseases on Medical channel
add: ALL scanners update medical records with diseases detected
/:cl:

### CODE REVIEW

**Comment by DerpiestBruh** [comment]
not a fan

File: code/game/objects/items/devices/scanners.dm [hunk 1]
```diff
@@ -369,11 +369,28 @@ GENE SCANNER
...
 	for(var/thing in M.diseases)
 		var/datum/disease/D = thing
 		if(!(D.visibility_flags & HIDDEN_SCANNER))
+			if(speaking_mob)
+				var/obj/item/radio/headset/radio = new(speaking_mob)
+				radio.keyslot = new /obj/item/encryptionkey/headset_med
+				radio.subspace_transmission = TRUE
+				radio.canhear_range = 0
+				radio.recalculateChannels()
+				radio.talk_into(speaking_mob, "Warning! Crewmember [M] has [D.form]: [D.name] ([D.stage]/[D.max_stages]). Possible cure is [D.cure_text]", RADIO_CHANNEL_MEDICAL)  // alert
```
Comment (zeskorion) [inline code/game/objects/items/devices/scanners.dm]: I like this idea, but i worry about it spamming the channel. this should have a cooldown

Comment (etherware-novice) [reply code/game/objects/items/devices/scanners.dm]: oh good idea

Comment (zeskorion) [reply code/game/objects/items/devices/scanners.dm]: may also be wise to broadcast transmission vectors!

**Comment by zeskorion** [comment]
this may seem hugboxy, but an advance warning to medical when a virus is about is a good way tiw arn them to start producing cures and quarantining

**Comment by etherware-novice** [comment]
whats hugboxy?

**Comment by DerpiestBruh** [comment]
I mean what stops a doctor from announcing by themselves that theres a disease?

**Comment by etherware-novice** [comment]
well yes but all the relevant info is transmitted automatically, plus it means non medical personell can still alert you of diseases

**Comment by Chessus** [comment]
Cool shit

**Comment by Fronsis** [comment]
Maybe move this feature to the scanner-gates? Bah, they're not called scanner gates, they're the one that are full blue with a human-sprite in the middle, at least that way they're a little more useful and will see more usage (though they still don't let mechas/borg go through them which is a pain)

**Comment by etherware-novice** [comment]
whats a scanner-gate?

**Comment by zeskorion** [comment]
> Maybe move this feature to the scanner-gates? Bah, they're not called scanner gates, they're the one that are full blue with a human-sprite in the middle, at least that way they're a little more useful and will see more usage (though they still don't let mechas/borg go through them which is a pain)

the holobarriers? they suck. wont quite work 
this works to prevent doctor negligence because no one knows how to handle viruses

**Comment by Penwin0** [comment]
This can get really spammy considering how many people can be using health analyzers at a time

**Comment by Kratairo** [comment]
So uh what is viro supposed to do when its already impossible to make a virus that spreads and has at least decent resistance (also why is it so extremely easy to cure a virus anyways, takes 10 seconds to make even the hardest cure)

**Comment by etherware-novice** [comment]
> This can get really spammy considering how many people can be using health analyzers at a time 

theres a cooldown

**Comment by zeskorion** [comment]
> So uh what is viro supposed to do when its already impossible to make a virus that spreads and has at least decent resistance (also why is it so extremely easy to cure a virus anyways, takes 10 seconds to make even the hardest cure)

not expect to kill literally everyone with a virus?
make corpses dangerous isntead of just using direct damage?

**Comment by etherware-novice** [comment]
yea, wat

**Comment by DerpiestBruh** [comment]
Making analyzers announce diseases and the cure together kills the few interactions between med staff in med radio channel and the department itself. And it makes medical (which is already ez) easier in a bad way, IMHO.

**Comment by zeskorion** [comment]
> Making analyzers announce diseases and the cure together kills the few interactions between med staff in med radio channel and the department itself. And it makes medical (which is already ez) easier in a bad way, IMHO.

noticing the diseases is hard enough to trust our player with, notifying them may at least let them get a quarantine in place first

**Comment by DerpiestBruh** [comment]
> noticing the diseases is hard enough to trust our player with

LOL

**Comment by zeskorion** [comment]
> > noticing the diseases is hard enough to trust our player with
> 
> > trust our player
> 
> LOL

minor spellng mistake

**Comment by PowerfulBacon** [comment]
Interactions and communications should be encouraged in the game, anything promoting players playing in a singleplayer style should be discouraged and goes against the design direction of the game. Closing for the stated reasons.

**Pull request closed without merging**